rfcs/OperationExpressions.md

```diff
@@ -0,0 +1,391 @@
+path with the operation type (you can do this for queries too, but just like in
+operation documents, the query keyword is optional):
+
+- `mutation>createUser>user>name` expands to
+  `mutation ($input: CreateUserInput!) { createUser(input: $input) { user { name } } }`
+- `subscription>currentUserUpdated>name` expands to
+  `subscription { currentUserUpdated { name } }`
+- `query>me>name` expands to `query { me { name } }`
+
+You may name operations by prefixing with an operation name followed by a colon;
+for example:
+
+- `MyQuery:>me>name` and `MyQuery:query>me>name` expand to
+  `query MyQuery { me { name } }`.
+- `MyMutation:mutation>createUser>name` expands to
+  `mutation MyMutation { createUser { name } }`.
+- `MySubscription:subscription>userCreated>name` expands to
+  `subscription MySubscription { userCreated { name } }`.
+
+#### Fragments
+
+Fragments start with a type name followed by a period: `User.friends>name`
+expands to `... on User { friends { name } }`.
+
+You can name fragments by prefixing with a fragment name and a colon:
+`FriendNames:User.friends>name` expands to
+`fragment FriendNames on User { friends { name } }`.
+
+Other examples:
+
+- `MyFragment:Node.User.fullName:name` expands to
+  `fragment MyFragment on Node { ... on User { fullName: name } }`
+
+- `MyQuery:>allEntities>edges>node>MyNodeFragment:Node.MyUserFragment:User.fullName:name`
+  expands to
+
+  ```graphql
+  query MyQuery {
+    allEntities {
+      edges {
+        node {
+          ...MyNodeFragment
+        }
+      }
+    }
+  }
+
+  fragment MyNodeFragment on Node {
+    ...MyUserFragment
+  }
+
+  fragment MyUserFragment on User {
+    fullName: name
+  }
+  ```
+
+#### Arguments
+
+Arguments use the same syntax as Schema Coordinates; namely parenthesis and a
+colon: `>searchBusinesses(name:)>city`.
+
+We also allow you to reference input objects used in arguments, for example:
+
+`>searchBusinesses(where.size.greaterThan:)>city`
+
+expands to something like:
+
+```graphql
+query($whereSizeGreaterThan: Int) {
+  searchBusinesses(where: { size: { greaterThan: $whereSizeGreaterThan } }) {
+    city
+  }
+}
+```
+
+Further we allow for multiple arguments to be specified, joined with commas:
+
+`>searchBusinesses(where.size.greaterThan:,where.size.lessThan:,where.city.equalTo:)>name`
+
+expands to something like:
+
+```graphql
+query(
+  $whereSizeGreaterThan: Int
+  $whereSizeLessThan: Int
+  $whereCityEqualTo: String
+) {
+  searchBusinesses(
+    where: {
+      size: { greaterThan: $whereSizeGreaterThan, lessThan: $whereSizeLessThan }
+      city: { equalTo: $whereCityEqualTo }
+    }
+  ) {
+    name
+  }
+}
+```
+
+> NOTE: the following number syntax probably needs more thought. Added only for
+> completeness.
+
+We also allow `[number]` syntax to refer to a numbered entry in a list, or `[]`
+to refer to the next entry; e.g.:
+
+`>findUsers(byIds[]:,byIds[],byIds[],byIds[5])>name`
```

Comment:
i'm a little confused here - are we implying that `byIds5` is an array?

(it's writen as a single `ID` type in the query below)

Reply:
The auto-variables may be adding some confusion here. The `byIds:` _argument_ is a list, so we can specify members of it; since they have no values if we want to have it converted to a document we need to pass something so we automatically generate variables for it; in this case `$byIds5` (an ID, not a list) is one of the members of the list that’s fed to `byIds:`